### PR TITLE
Refacor dimentionality reduction into a class.

### LINKED ISSUE #82: Channel_wise toeplitz not working
`model.svd_fns` is returning a `2D` tensor while `corevectors.dimReduction.svds.conv2d_toeplitz_svd_projection` expects a `3D` tensor. `coreVectors` computation works, but break later on `peepholes` .

### LINKED ISSUE #111: Remove `'peepholes'` key
Peepholes are saved under ['peepholes'][<layer>]; however, the first key is no longer needed and should be removed.

### LINKED ISSUE #117: SVD reduction functions optimization
The SVD reduction functions on `corevectors/dimReduction/svds.py` send the `reduct_m` matrices to the device every batch. It would be better to wrap those in a class with a `__call__` method, which is set with the appropriate method for each SVD type on construction.

### PR DESCRIPTION
**Pull request by lsrosa**
Dimensionality reduction is now wrapped within classes. The classes inherit `coreVectors/dimReduction/dim_reduction_base.DimReductionBase` class, forcing the implementation of a `__call__()` method (equivalent to applying the svd projections, or avgPooling).

List of changes:
- depracate use_s and channel_wise
- refactor linear svd
- refactor conv2d toepliz svd
- refactor conv2d kernel svd
- refactor avgPooling
- refactor vit linear svd
- refactor vit token dim reductions
- refactor peepholes and drillers
- remove driller's suffix
- remove peepholes key from phs

### CODE REVIEW

File: peepholelib/coreVectors/dimReduction/svds/conv2d_kernel_svd.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        - act_data (torch.tensor): batched input activations
+        
+        Returns:
+        - cvs (torch.tensor) = batched projected activations
+        '''
+        act_data = kwargs['act_data'] 
+        n_act = act_data.shape[0]
+        unrolled_acts = unroll_conv2d_activations(acts=act_data, layer=self.layer)
+        cvs = (self.reduct_m@unrolled_acts).transpose(1, 2)
+        print(cvs.shape)
```
Comment (lorenzoCapelli) [inline peepholelib/coreVectors/dimReduction/svds/conv2d_kernel_svd.py]: leftover

File: peepholelib/coreVectors/dimReduction/svds/linear_svd.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        # get ref for the layer
+        _layer = model._target_modules[layer]
+        device = model.device
+
+        if file_path.exists():
+            if verbose: print(f'File {file_path} exists. Loading from disk.')
+            self._svd = torch.load(file_path)
+        else: 
+            # Turn on activation saving
+            model.set_activations(save_input=True, save_output=False)
```
Comment (lorenzoCapelli) [inline peepholelib/coreVectors/dimReduction/svds/linear_svd.py]: remove set_activations

File: peepholelib/coreVectors/dimReduction/svds/vit_linear_svd.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            U, s, Vh = torch.svd_lowrank(W, q)
+            U, s, Vh = U.detach().cpu(), s.detach().cpu(), Vh.detach().cpu()
+            self._svd = {
+                    'U': U,
+                    's': s,
+                    'Vh': Vh.T
+                    }
+
+            # Turn off activation saving
+            model.set_activations(save_input=False, save_output=False)
```
Comment (lorenzoCapelli) [inline peepholelib/coreVectors/dimReduction/svds/vit_linear_svd.py]: remove set_activations

File: peepholelib/peepholes/classifiers/tgmm.py [hunk 4]
```diff
@@ -75,9 +75,8 @@ def classifier_probabilities(self, **kwargs):
         '''
         
         data = kwargs['data']
-
+        print('Prob: ', data)
```
Comment (lorenzoCapelli) [inline peepholelib/peepholes/classifiers/tgmm.py]: leftover

File: peepholelib/coreVectors/dimReduction/svds/linear_svd.py (hunk 2 above)
Comment (lsrosa) [reply peepholelib/coreVectors/dimReduction/svds/linear_svd.py]: cleaned

**Review by lorenzoCapelli — APPROVED** [review]
